packages/clerk-js/src/ui/elements/Menu.tsx

```diff
@@ -0,0 +1,175 @@
+  const { popoverCtx } = useMenuState();
+  const { reference, toggle } = popoverCtx;
+
+  return (
+    <Button
+      ref={reference}
+      onClick={toggle}
+      {...props}
+    />
+  );
+};
+
+const findMenuItem = (el: Element, siblingType: 'prev' | 'next', options?: { countSelf?: boolean }) => {
+  let tagName = options?.countSelf ? el.tagName : '';
+  let sibling: Element | null = el;
+  while (sibling && tagName !== 'BUTTON') {
+    sibling = sibling[siblingType === 'prev' ? 'previousElementSibling' : 'nextElementSibling'];
+    tagName = sibling?.tagName ?? '';
+  }
+  return sibling;
+};
+
+type MenuListProps = PropsOfComponent<typeof Col>;
+
+export const MenuList = (props: MenuListProps) => {
+  const { sx, ...rest } = props;
+  const { popoverCtx } = useMenuState();
+  const { floating, styles, isOpen } = popoverCtx;
+  const containerRef = useRef<HTMLDivElement>(null);
+
+  useEffect(() => {
+    const current = containerRef.current;
+    floating(current);
+    current?.focus();
+  }, [isOpen]);
+
+  const onKeyDown = (e: React.KeyboardEvent<HTMLDivElement>) => {
+    const current = containerRef.current;
+    if (!current) {
+      return;
+    }
+
+    if (current !== document.activeElement) {
+      return;
+    }
+
+    if (e.key === 'ArrowDown') {
+      return (findMenuItem(current.children[0], 'next', { countSelf: true }) as HTMLElement)?.focus();
+    }
+  };
+
+  if (!isOpen) {
+    return null;
+  }
+
+  return (
+    <Portal>
+      <Col
+        ref={containerRef}
+        role='menu'
+        onKeyDown={onKeyDown}
+        tabIndex={0}
+        sx={[
+          theme => ({
+            backgroundColor: colors.makeSolid(theme.colors.$colorBackground),
+            border: theme.borders.$normal,
+            outline: 'none',
+            borderRadius: theme.radii.$lg,
+            borderColor: theme.colors.$blackAlpha200,
+            paddingTop: theme.space.$2,
+            paddingBottom: theme.space.$2,
+            overflow: 'hidden',
+            top: `calc(100% + ${theme.space.$2})`,
+            animation: `${animations.dropdownSlideInScaleAndFade} ${theme.transitionDuration.$slower} ${theme.transitionTiming.$slowBezier}`,
+            transformOrigin: 'top center',
+            boxShadow: theme.shadows.$boxShadow1,
+            zIndex: theme.zIndices.$dropdown,
+          }),
+          sx,
+        ]}
+        style={styles}
+        {...rest}
+      />
+    </Portal>
+  );
+};
+
+type MenuItemProps = PropsOfComponent<typeof Button>;
+
+export const MenuItem = (props: MenuItemProps) => {
+  const { sx, onClick, ...rest } = props;
+  const { popoverCtx } = useMenuState();
+  const { toggle } = popoverCtx;
+  const buttonRef = useRef<HTMLButtonElement>(null);
+
+  const onKeyDown = (e: React.KeyboardEvent<HTMLButtonElement>) => {
+    const current = buttonRef.current;
+    if (!current) {
+      return;
+    }
+
+    const keyMap: Record<string, 'prev' | 'next'> = {
```

Comment:
Is there a reason for this very generic `keyMap` instead of an early return if `e.key` is not ArrowUp or ArrowDown?

If you think we'll need to detect more events in the future, consider using a switch on `e.key` instead

Reply:
Will change it. It looks better.